src/main/java/org/bitsofinfo/hazelcast/discovery/docker/swarm/SwarmDiscoveryUtil.java

```diff
@@ -397,6 +397,21 @@ private boolean isSelf(NetworkAttachment networkAttachment) throws SocketExcepti
 
 		logger.info("Number of services matching given criteria = " + services.size());
 
+		if (true || services.size() == 0) {
+			// TODO remove true once log output has been verified via Travis CI
+			List<Service> allServices = docker.listServices();
+
+			StringBuilder sb = new StringBuilder();
+			String delim = "";
+			for (Service s : allServices) {
+				sb.append(delim).append(s.spec().name());
+				delim = ",";
+			}
+
+			logger.info("No service match for given criteria");
```

Comment:
You don't need the verbosity of 2 log statements here, just combine into one one

This entire block needs to be configured by a new configuration option. OFF by default

Reply:
I was planning to drop the 2nd to DEBUG, hence having two.  DEBUG is off by default.

Reply:
Also, withdrawing this PR in order to resubmit with the TODO item addressed.